test_runner/fixtures/neon_fixtures.py

```diff
@@ -3458,12 +3460,34 @@ def create(
         self.pgdata_dir = self.env.repo_dir / path
         self.logfile = self.endpoint_path() / "compute.log"
 
-        config_lines = config_lines or []
+        if use_lfc is None:
+            use_lfc = not NO_DEFAULT_LFC
 
         # set small 'max_replication_write_lag' to enable backpressure
         # and make tests more stable.
         config_lines = ["max_replication_write_lag=15MB"] + config_lines
 
+        # Delete file cache if it exists (and we're recreating the endpoint)
+        if use_lfc:
+            if (lfc_path := Path(self.lfc_path())).exists():
+                lfc_path.unlink()
+            else:
+                lfc_path.parent.mkdir(parents=True, exist_ok=True)
+
+            config_lines = [
+                "shared_buffers = 512kB",
+                f"neon.file_cache_path = '{self.lfc_path()}'",
+                "neon.max_file_cache_size = 512kB",
+                "neon.file_cache_size_limit = 512kB",
+            ] + config_lines
+        else:
+            config_lines = [
+                line
+                for line in config_lines
+                if not line.startswith("neon.max_file_cache_size")
+                and not line.startswith("neon.file_cache_size_limit")
+            ]
```

Comment:
I don't understand why do we need this else block

Reply:
Sometimes we need to specify a custom size of the LFC.
We need to remove `neon.max_file_cache_size` and `neon.file_cache_size_limit` from the config if we don't use the LFC.

Reply:
I understand both statements, but I don't understand them together in the context of this part of the code 🤣 

Where do we set `neon.max_file_cache_size` / `neon.file_cache_size_limit` for tests with enabled LFC (so we need to delete it here, I expect `use_lfc` is `True` for them)?

Reply:
Test developers can set these values to specify the custom size for LFC. 
E.g. https://github.com/neondatabase/neon/blob/8fe0257ca7c38fd4cb3be34276c48e14f3d0d17e/test_runner/regress/test_logical_replication.py#L584-L590
It will raise an error with the default size of LFC, so we have to specify the values here.
On the other hand, if we want to run without LFC, we have to remove them.

Reply:
I suppose for that test `USE_LFC=True` and we're not entering `else`-block, isn't it?
Or why do we set LFC-related parameters for tests with disabled LFC?

Reply:
It can be `true` or `false` depending on the `USE_LFC` environment variable.
If `USE_LFC` is `true`, we don't enter the else block
If it's false, we have to remove the LFC-related settings.